File: ssm-sdk-client/src/main/java/ssm/client/Utils/JsonUtils.java
package ssm.client.Utils;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.PropertyAccessor;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.module.kotlin.KotlinModule;

import java.io.IOException;
import java.util.Iterator;
import java.util.List;
import java.util.concurrent.CompletionException;
import java.util.function.Function;

public class JsonUtils {

    private static final ObjectMapper mapper;

    static {
        mapper = new ObjectMapper()
                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
                .setSerializationInclusion(JsonInclude.Include.NON_NULL)
                .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY)
                .registerModule(new KotlinModule());
    }

    public static String toJson(Object obj) throws JsonProcessingException {
        mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
        return mapper.writeValueAsString(obj);
    }

    public static <T> T toObject(String value, Class<T> clazz) throws IOException {
        mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
        return mapper.readValue(value, clazz);
    }

    public static <T> T toObject(String value, TypeReference<T> clazz) throws IOException {
        mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
        return mapper.readValue(value, clazz);
    }

}

File: ssm-sdk/ssm-sdk-json/src/main/kotlin/ssm/sdk/json/JsonUtils.kt
package ssm.sdk.json

import com.fasterxml.jackson.annotation.JsonAutoDetect
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.PropertyAccessor
import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.DeserializationFeature
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.KotlinModule
import java.io.IOException
import java.io.Reader

object JsonUtils {

	private val mapper: ObjectMapper = ObjectMapper()
		.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
		.setSerializationInclusion(JsonInclude.Include.NON_NULL)
		.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY)
		.registerModule(KotlinModule())

	@Throws(JsonProcessingException::class)
	fun <T> toJson(obj: T): String {
		return mapper.writeValueAsString(obj)
	}

	@Throws(IOException::class)
	fun <T> toObject(value: String, clazz: Class<T>): T {
		return mapper.readValue(value, clazz)
	}

	@Throws(IOException::class)
	fun <T> toObject(value: Reader, clazz: Class<T>): T {
		return mapper.readValue(value, clazz)
	}

	@Throws(IOException::class)
	fun <T> toObject(value: String, clazz: TypeReference<T>): T {
		return mapper.readValue(value, clazz)
	}
}
